sdk/src/main/java/io/coti/sdk/TransactionCreation.java

```diff
@@ -45,49 +49,66 @@ public TransactionData createTransferTransaction(BigDecimal amount, String trans
             throw new BalanceException(Constants.INSUFFICIENT_FUNDS_MESSAGE);
         }
         BaseTransactionCreation baseTransactionCreation = new BaseTransactionCreation(nativeCurrencyHash, fullNodeAddress, trustScoreAddress);
-        List<BaseTransactionData> baseTransactions = baseTransactionCreation.createBaseTransactions(userPrivateKey, senderHash, amount, addressHash, feeIncluded, receiverAddress);
-        CryptoUtils.createAndSetBaseTransactionsHash(baseTransactions);
-        GetTransactionTrustScoreResponse trustScoreResponse = getTrustScore(baseTransactions, userPrivateKey, senderHash);
-        double trustScore = 0;
-        TransactionTrustScoreData transactionTrustScoreData = new TransactionTrustScoreData(trustScore);
-        if (trustScoreResponse.getTransactionTrustScoreData() != null) {
-            transactionTrustScoreData = Mapper.map(trustScoreResponse.getTransactionTrustScoreData()).toTrustScoreData();
-            trustScore = trustScoreResponse.getTransactionTrustScoreData().getTrustScore();
-        }
-        TransactionData transactionData = createTransferTransactionData(baseTransactions, transactionDescription, trustScore, addressHash);
-        transactionData.setTrustScoreResults(Collections.singletonList(transactionTrustScoreData));
-        transactionData.setSenderHash(senderHash);
-        transactionData.setSenderSignature(CryptoUtils.signTransactionData(transactionData, userPrivateKey));
+        List<BaseTransactionData> baseTransactions = baseTransactionCreation.createBaseTransferTransactions(userPrivateKey, senderHash, amount, addressHash, feeIncluded, receiverAddress);
+        return createTransactionDataByType(transactionDescription, baseTransactions, TransactionType.Transfer);
+    }
 
+    public TransactionData createTokenTransactionByType(BaseTransactionData tokenFeeBT, FullNodeFeeResponse fullNodeFeeResponse,
```

Comment:
we have a specific method to create transfer transaction. please create a method for token minting and token generation